packages/engine/src/worker/runner/javascript/mod.rs

```diff
@@ -1257,26 +1665,30 @@ impl<'m> RunnerImpl<'m> {
             state_group_start_indices,
         } = ctx_batch_sync;
 
-        let args = mv8::Values::from_vec(vec![
-            sim_id_to_js(mv8, sim_run_id),
-            batch_to_js(
-                mv8,
-                context_batch.segment().memory(),
-                context_batch.segment().persisted_metaversion(),
-            )?,
-            idxs_to_js(mv8, &state_group_start_indices)?,
-            current_step_to_js(mv8, current_step),
-        ]);
-        let _: mv8::Value<'_> = self
-            .embedded
+        let js_sim_id = sim_id_to_js(scope, sim_run_id);
+        let js_batch_id = batch_to_js(
+            scope,
+            context_batch.segment().memory(),
+            context_batch.segment().persisted_metaversion(),
+        )?;
+        let js_idxs = new_js_array_from_usizes(scope, &state_group_start_indices)?;
```

Comment:
Do we get indices or an array here?

Reply:
It's an array of indices

Reply:
Even though it's a bit verbose, could we change the name to `js_state_group_start_indices` or similar then?

Reply:
It's called twice, once for the group start indices and another time for the batch indices, are these the same?

Reply:
I'm pretty sure they are. @Alfred-Mountfield can you confirm? 

Related: [Figure out Batches vs Groups vs Pools](https://app.asana.com/0/1199548034582004/1201800853370429) (_internal_)

Reply:
🙃 Most of the time they are, but as we established the message and state batches don't correspond to groups in the context..

I don't think there's much harm in the function being called what it is. It creates a JS array from a set of usizes. The context about building indices is at the level that calls the function.